mayastor-test/e2e/install/install_test.go

```diff
@@ -0,0 +1,138 @@
+package basic_test
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"path"
+	"runtime"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/deprecated/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var cfg *rest.Config
+var k8sClient client.Client
+var k8sManager ctrl.Manager
+var testEnv *envtest.Environment
+
+// Helper for passing yaml to kubectl
+func applyYaml(rootDir string, filename string) {
```

Comment:
One of the reasons I suggested to make use of the k8s ecosystem is "k8s.io/kubernetes/test/e2e/framework" which instead of doing our own thing, allows replacing this with:

```go
framework.RunKubectl("-n", nameSpace, "delete", "-f", configMapPath)
```

https://pkg.go.dev/k8s.io/kubernetes/test/e2e/framework provides a better overview.

question: Is there a reason we are *not* using that right now?

Reply:
e2e/framework is part of k8s.io/kubernetes and isn't intended to be used as a standalone library. Seemingly by design, it therefore can't be imported as a go module https://github.com/kubernetes/kubernetes/issues/79384#issuecomment-505725449 (referenced closer to home https://github.com/openebs/zfs-localpv/pull/131).
I can hack it in, but I think that RunKubectl is the only function in that library that we'd really want at this point, and it's ~20 lines of mess in the mod files to save 1 line in the code.